azrcrv-redirect.php

```diff
@@ -541,10 +546,13 @@ function display_manage_redirects() {
 
 				$id = (int) $_POST['id'];
 
-				// "complex" placeholder used as string cannot be wrapped in single quotes.
+				// "complex" placeholder used intentionally so that table name
+				// is not wrapped in single quotes.
+				// phpcs:ignore WordPress.DB.PreparedSQLPlaceholders.UnquotedComplexPlaceholder
 				$sql = $wpdb->prepare( 'SELECT source_url,redirect_type,destination_url FROM %1s WHERE id = %d LIMIT 0,1', $tablename, $id );
 
-				// wpdb->prepare used above.
+				// $wpdb->prepare used above.
+				// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
 				$row = $wpdb->get_row( $sql );
```

Comment:
All of the DB queries are done this way, but these checks disappear if they are done the "standard" way instead:

```php
$row = $wpdb->get_row( $wpdb->prepare( 'SELECT source_url,redirect_type,destination_url FROM {$wpdb->prefix}azrcrv_redirects WHERE id = %d LIMIT 0,1', $id ) );
```

The table name is currently determined as follows:

```php
const DATABASE_TABLE = 'azrcrv_redirects';
// [...]
$tablename = $wpdb->prefix . DATABASE_TABLE;
```

However, I don't think this approach provides much actual benefit. Aside from being non-standard, the database name cannot easily be changed (you'd have to add a migration routine to the plugin), and if you do need to find all the places where the table name is used, then doing a text search for a string like `prefix}azrcrv_redirects` across the entire codebase is actually even easier than deciphering the two-step table name computation in use today.

Similarly to my above comment, I think whether this is _modified_ or _annotated with comments and `phpcs:ignore`_ should be left up to the plugin author, but it definitely makes reviews more difficult. For example, I didn't know that `%1$s` ("complex" placeholders) had different behavior in `$wpdb->prepare` until I came across this code, so other reviewers probably wouldn't either.